reana_job_controller/compute4punch_job_manager.py

```diff
@@ -0,0 +1,366 @@
+# This file is part of REANA.
+# Copyright (C) 2019, 2020, 2021, 2022, 2023, 2024 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+import base64
+import logging
+import os
+import re
+
+from paramiko.sftp_client import SFTPClient
+from stat import S_ISDIR
+from typing import Iterable
+
+from reana_commons.workspace import is_directory, open_file, walk
+from reana_job_controller.job_manager import JobManager
+from reana_job_controller.utils import SSHClient, motley_cue_auth_strategy_factory
+from reana_job_controller.config import (
+    C4P_LOGIN_NODE_HOSTNAME,
+    C4P_LOGIN_NODE_PORT,
+    C4P_SSH_TIMEOUT,
+    C4P_SSH_BANNER_TIMEOUT,
+    C4P_SSH_AUTH_TIMEOUT,
+)
+
+
+class Compute4PUNCHJobManager(JobManager):
+    """Compute4PUNCH Job Manager"""
+
+    C4P_WORKSPACE_PATH = ""
+    """Absolute path on the Compute4PUNCH head node used for submission"""
+    C4P_HOME_PATH = ""
+    """Default Compute4PUNCH home directory"""
+    SUBMIT_ID_PATTERN = re.compile(r"Proc\s(\d+\.\d+)")
+    """ regex to search the Job ID in a submit Proc line """
+
+    def __init__(
+        self,
+        docker_img=None,
+        cmd=None,
+        prettified_cmd=None,
+        env_vars=None,
+        workflow_uuid=None,
+        workflow_workspace=None,
+        cvmfs_mounts="false",
+        shared_file_system=False,
+        job_name=None,
+        c4p_cpu_cores=None,
+        c4p_memory_limit=None,
+        c4p_additional_requirements=None,
+        **kwargs,
+    ):
+        """
+        Compute4PUNCH Job Manager.
+
+        :param docker_img: Docker image.
+        :type docker_img: str
+        :param cmd: Command to execute.
+        :type cmd: list
+        :param prettified_cmd: prettified version of command to execute.
+        :type prettified_cmd: str
+        :param env_vars: Environment variables.
+        :type env_vars: dict
+        :param workflow_uuid: Unique workflow id.
+        :type workflow_uuid: str
+        :type workflow_workspace: str
+        :param cvmfs_mounts: list of CVMFS mounts as a string.
+        :type cvmfs_mounts: str
+        :param shared_file_system: if shared file system is available.
+        :type shared_file_system: bool
+        :param job_name: Name of the job
+        :type job_name: str
+        :param c4p_cpu_cores: number of CPU cores to use on C4P
+        :type c4p_cpu_cores: int, str
+        :param c4p_memory_limit: maximum memory to be used on C4P
+        :type c4p_memory_limit: int, str
```

Comment:
Why both `int` and `str`?

Reply:
I think `str` should be fine. Fixed in https://github.com/reanahub/reana-job-controller/pull/430/commits/c6b53c655e3bf242a9accf3d0e90816f9e08c466.